File: API-MUSICSHOP/models/instrumento.ts
import mongoose, { Schema, Document } from "mongoose";
import IInstrumento from "../interfaces/instrumento";

const SchemaInstrumento: Schema = new Schema<IInstrumento>(
  {
    nombre: { type: Schema.Types.String, required: true },
    descripcion: { type: Schema.Types.String, required: true },
    disponibilidad: { type: Schema.Types.Boolean,required: true },
    precio: {type: Schema.Types.Number ,required: true},
   
  },
  { timestamps: false, versionKey: false }
);

export const InstrumentoModel = mongoose.model<IInstrumento & Document>(
  "instrumentos",
  SchemaInstrumento
);

File: AplicacionMusica/app/src/main/java/com/example/aplicacionmusica/modelo/Instrumento.kt
package com.example.aplicacionmusica.modelo

data class Instrumento(
    val nombre: String,
    val descripcion: String,
    val disponibilidad: Boolean,
    val precio: Number,

) {
    val _id: String = ""

    constructor() : this("", "", false,  0)
    
}
